codelists/scripts/import_codelists_from_xlsx.py

```diff
@@ -0,0 +1,333 @@
+        return codelist_name_aliases.get(name, name)
+
+    aliased_names = codelist_df["codelist_name"].apply(update_name)
+    codelist_df["codelist_name"] = aliased_names
+
+    return codelist_df
+
+
+def get_post_data(config, codelist_df, create_new_codelist, force_new_version):
+    """
+    Return the relevant data to post to the api endpoint to create a new
+    codelist or codelist version
+    """
+    first_row = codelist_df.iloc[0]
+    codelist_name = first_row["codelist_name"]
+    coding_system_from_data = first_row["coding_system"]
+    coding_system_id = config["coding_systems"][coding_system_from_data]["id"]
+    release_db_alias = config["coding_systems"][coding_system_from_data]["release"]
+    tag = config.get("tag")
+
+    post_data = {"coding_system_database_alias": release_db_alias, "tag": tag}
+    if coding_system_id in CODING_SYSTEMS_WITH_OLD_STYLE_CODELISTS:
+        # create an old-style codelist/version with csv_data
+        post_data["csv_data"] = codelist_df[["code", "term"]].to_csv()
+    else:
+        post_data.update(
+            {
+                "always_create_new_version": force_new_version,
+                "codes": list(set(codelist_df["code"])),
+            }
+        )
+
+    if create_new_codelist:
+        post_data.update(
+            {
+                "name": codelist_name,
+                "coding_system_id": coding_system_id,
+                "description": None,
+                "methodology": None,
+            }
+        )
+
+    return codelist_name, coding_system_id, post_data
+
+
+class ValidateHost(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        if values != "bar":
+            parser.error(f"Please enter a valid. Got: {values}")
+        setattr(namespace, self.dest, values)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("xlsx_file", type=Path, help="Path/to/xlsx/file")
+    parser.add_argument("config_file", type=Path, help="Path/to/config/json/file")
+    parser.add_argument(
+        "--force-new-version",
+        "-f",
+        action="store_true",
+        help="Always create a new version, even if a version with identical codes already exists.",
+    )
+    parser.add_argument(
+        "--live-run",
+        action="store_true",
+        help=(
+            "Run this command for real; if not specified, this will be a dry run that reports "
+            "codelists/versions that will be created but does not actually do anything."
+        ),
+    )
+    parser.add_argument(
+        "--host",
+        default="http://localhost:8000",
+        help="Host to use for API calls; https://opensafely.org for live prod run",
+    )
+
+    arguments = parser.parse_args()
+    config = read_config(arguments.config_file)
+
+    return (
+        arguments.xlsx_file,
+        config,
+        arguments.host,
+        arguments.force_new_version,
+        not arguments.live_run,
+    )
```

Comment:
Perhaps worth adding a test that this is not run live by default.  How paranoid are we feeling?